crates/shared/src/sources/uniswap_v3/pool_fetching.rs

```diff
@@ -147,10 +145,21 @@ impl PoolsCheckpointHandler {
             pools_by_token_pair.entry(pair).or_default().insert(pool.id);
         }
 
+        // can't fetch the state of all pools in constructor for performance reasons,
+        // so let's fetch the top `max_pools_to_initialize_cache` pools with the highest
+        // liquidity
+        registered_pools.pools.sort_unstable_by(|a, b| {
+            a.total_value_locked_eth
+                .partial_cmp(&b.total_value_locked_eth)
+                .unwrap()
+        });
         let pool_ids = registered_pools
             .pools
-            .iter()
+            .clone()
```

Comment:
You don't have to clone the whole thing. You can do..iter().rev().map(value.clone())